packages/api/src/handlers/job.ts

```diff
@@ -0,0 +1,24 @@
+import { BullBoardRequest, ControllerHandlerReturnType, QueueJob } from '../../typings/app';
+import { queueProvider } from '../providers/queue';
+import { jobProvider } from '../providers/job';
+
+async function getJobState(
+  _req: BullBoardRequest,
+  job: QueueJob
+): Promise<ControllerHandlerReturnType> {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const state = await job?.getState();
```

Comment:
I'm not 100% sure what to do here. Exposing `QueueJob#getState(): Promise<Status>;` in `typings/app.ts` causes a big conflict between `BullAdapter` and `BaseAdapter`.